### PR TITLE
[14.0][MIG] account_payment_term_extension: Migration to 14.0

### PR DESCRIPTION
**Pull request by pedrobaeza**
@Tecnativa TT27702

### CODE REVIEW

**Comment by pedrobaeza** [comment]
@alexis-via do you give your blessing?

File: account_payment_term_extension/models/account_payment_term.py [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+        date_ref = date_ref or fields.Date.today()
+        amount = value
+        result = []
+        if not currency:
+            if self.env.context.get("currency_id"):
+                currency = self.env["res.currency"].browse(
+                    self.env.context["currency_id"]
+                )
+            else:
+                currency = self.env.user.company_id.currency_id
```
Comment (alexis-via) [inline account_payment_term_extension/models/account_payment_term.py]: Use self.env.company instead of self.env.user.company_id ?

Comment (pedrobaeza) [reply account_payment_term_extension/models/account_payment_term.py]: Changed, thanks.

Comment (joao-p-marques) [reply account_payment_term_extension/models/account_payment_term.py]: I think you used `self.env.user.company` instead which is causing an error in the tests: https://travis-ci.com/github/OCA/account-payment/jobs/472291835#L1181

Comment (pedrobaeza) [reply account_payment_term_extension/models/account_payment_term.py]: Indeed, I made the change blindly and very late, hehe. Now it should be OK.

File: account_payment_term_extension/models/account_payment_term.py [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+                )
+
+
+class AccountPaymentTermLine(models.Model):
+    _inherit = "account.payment.term.line"
+
+    amount_round = fields.Float(
+        string="Amount Rounding",
+        digits="Account",
+        # TODO : I don't understand this help msg ; what is surcharge ?
```
Comment (joao-p-marques) [inline account_payment_term_extension/models/account_payment_term.py]: Any chance this `TODO` comment cant be resolved?

File: account_payment_term_extension/tests/test_account_payment_term_multi_day.py [hunk 3]
```diff
@@ -0,0 +1,322 @@
+import odoo.tests.common as common
+from odoo import fields
+from odoo.exceptions import ValidationError
+from odoo.tests.common import Form
+
+
+class TestAccountPaymentTermMultiDay(common.TransactionCase):
+    def setUp(self):
+        super(TestAccountPaymentTermMultiDay, self).setUp()
```
Comment (joao-p-marques) [inline account_payment_term_extension/tests/test_account_payment_term_multi_day.py]:
```suggestion
        super().setUp()
```

File: account_payment_term_extension/tests/test_payment_term.py [hunk 4]
```diff
@@ -0,0 +1,162 @@
+# Copyright 2015-2016 Akretion
+# (Alexis de Lattre <alexis.delattre@akretion.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo.exceptions import ValidationError
+from odoo.tests.common import TransactionCase
+
+
+class TestAccountPaymentTerm(TransactionCase):
+    def setUp(self):
+        super(TestAccountPaymentTerm, self).setUp()
```
Comment (joao-p-marques) [inline account_payment_term_extension/tests/test_payment_term.py]:
```suggestion
        super().setUp()
```

**Review by joao-p-marques — APPROVED** [review]
LGTM :+1: Just some small comments, not really important.

File: account_payment_term_extension/models/account_payment_term.py (hunk 1 above)
Comment (joao-p-marques) [reply account_payment_term_extension/models/account_payment_term.py]: Hehe ok :+1:

File: account_payment_term_extension/models/account_payment_term.py (hunk 2 above)
Comment (pedrobaeza) [reply account_payment_term_extension/models/account_payment_term.py]: Removed that part of the help message that doesn't apply here. I think it's copied from a field in Odoo (from pricelists).

**Comment by pedrobaeza** [comment]
@joao-p-marques Applied your suggestions and converted tests to SavepointCase for more optimization.

**Review by emagdalenaC2i — APPROVED** [review]
LGTM

**Pull request merged**